### ISSUE TITLE
Forums search only displays one page of results, and is missing filter options

### ISSUE THREAD

**Issue by baka-kaba** [Awful/Awful.apk#549]
When you perform a search, the site returns a page of results, with a bunch of linked pages of other results. We only display the first one, with no way to look through the rest, which limits its usefulness.

It should perform the search, then store that data so the user can access the rest of the query results without resubmitting. It would definitely be a good idea to cache viewed pages (if only while the search activity is open - watch out for app restarts though) and maybe even pull them all in in advance

We could maybe use the ``PageBar`` widget to jump between results pages, or add an infinite scrolling list and let it load in each page as the user scrolls down - scrolling is maybe the better use case (it's a static result and people aren't going to be jumping to specific pages, right?) but it could get annoying with lots of results. Might be worth a try though

There's also a couple of special search filters missing: ``since:`` and ``before:``
They look a bit weird and non-specific (using natural language) but might be worth including them for completeness, if anyone wants to use them they can work it out themselves

**Comment by Sereri** [comment]
Further pages are loaded via P2R at the bottom. Although that might be not that intuitive I guess.

When implementing search I went without pagination because there's no real reason to go to page XY directly. You don't know whats on page XY and have to go through every page looking for what you want anyway. Since that only left 'go to next page' I went with P2R since it's already a pattern in the app.

**Comment by baka-kaba** [comment]
Ohhh ok - I saw the P2R wrapper and thought it was a refresh, and swiping from the top did nothing so I thought maybe it was legacy code. Should have looked closer!

In that case it works already, it's like the infinite scroll only you have to drag to load the next set. I think automatically loading the next set would probably be better, just as an ease-of-use thing and you don't need to think about P2R (I didn't consider it at all for what it's worth, heh). But it's not a big deal so long as the functionality's there

Might be worth showing the result count though, it's nice to know what you're getting yourself into, if you're near the end, whether it's worth retrying with better terms etc

**Comment by Sereri** [comment]
Yeah I guess infinite scrolling or a load more button might be better. And a Found X pages or something at the top.
